ffi/src/primitives.rs

```diff
@@ -133,5 +141,153 @@ primitive_derive! { u32, i32, u64, i64 }
 #[cfg(not(target_family = "wasm"))]
 primitive_derive! { u8, i8, u16, i16 }
 
-// TODO: u128/i128 is not FFI-safe. Must be properly serialized!
-primitive_derive! { u128, i128 }
+/// Ffi-safe representation of `u128`
+#[derive(Clone, Copy, Debug, Default)]
+#[repr(transparent)]
+pub struct FfiU128(FfiTuple2<u64, u64>);
+
+/// Transparent to `FfiTuple<u64, u64>` which is `ReprC`
+unsafe impl ReprC for FfiU128 {}
+
+/// Ffi-safe representation of `i128`
+#[derive(Clone, Copy, Debug, Default)]
+#[repr(transparent)]
+pub struct FfiI128(FfiTuple2<u64, u64>);
+
+/// Transparent to `FfiTuple<u64, u64>` which is `ReprC`
+unsafe impl ReprC for FfiI128 {}
+
+macro_rules! int128_derive {
+    ($($src:ty => $dst:ty),+$(,)?) => {$(
+        impl Ir for $src {
+            type Type = Self;
```

Comment:
why didn't you use:

```suggestion
            type Type = Transparent;
```

or something like:
```rs
unsafe impl Transparent for $src {
    type Target = $dst;

    validation_fn=unsafe {$validity_fn},
    niche_value=$niche_val
}
```

Reply:
I afraid that if other side will use differently aligned `u128` values and transmuting pointer to such value would cause UB, because pointer isn't aligned properly. 
But i might be wrong.

Reply:
what you are transmuting is `FfiU128` into `FfiTuple2`. I don't see a problem there

Reply:
you should use this:
```rs
unsafe impl Transparent for $src {
    type Target = $dst;

    validation_fn=unsafe {$validity_fn},
    niche_value=$niche_val
}
```

Reply:
actually you can use `derive(FfiType)`, but you have to add `extern crate self as iroha_ffi`

Reply:
I'm wrong